### PR TITLE
feat(metrics): add pass@k / pass^k estimators and rows summarizer

### PR DESCRIPTION
**Pull request by isadominguez314**
### What

Adds `devops_bench/metrics/passk.py`: pure, dependency-free pass@k / pass^k calculators plus a `rows.json` summarizer CLI, with unit tests covering the math (part of #92).

- Unbiased estimators over `n` observed attempts with `c` passes:
  - `pass@k = 1 − C(n−c, k) / C(n, k)` — at least one of k attempts passes
  - `pass^k = C(c, k) / C(n, k)` — all k attempts pass (consistency)
- **One pass rule for the whole family**: a pass is `outcomeScore >= PASSK_THRESHOLD` (default **1.0**, a perfect composite). `passAt1` — the k = 1 case, which collapses to `c/n` — is reported alongside `passAtK`/`passPowK`, so pass@1 and pass@5 are the same formula at different k. Both `PASSK_THRESHOLD` and `DEFAULT_K = 5` are plain module constants.
- With fewer than k scored attempts a metric reports `null` — no estimate rather than an extrapolation from too few samples.
- `summarize_rows()` groups attempts by `(setupId, taskFolder)` across distinct `(runId, iteration)` pairs, dedupes retried uploads by the full doc id (distinct runs are never collapsed — they are the samples), and excludes unscored attempts from both `n` and `c`. Per-setup overall = mean over reporting tasks.
- CLI: `python -m devops_bench.metrics.passk RESULTS_ROOT` scans for `rows.json` files and writes `passk_summary.json`. Reuses `results/aggregate.py`'s row discovery; its private row loader is promoted to public `load_rows()`.

### Scope notes

- **Deferred by design** (per issue triage): harness support for launching k repeated/concurrent attempts. These metrics are computed from the scores of runs that already exist — k attempts arrive today as k separate runs of the same setup × task.
- Nothing in the harness calls this module yet; it lands here for review and visibility, as the reference implementation of the pass-family definitions. The leaderboard-side enablement (gke-labs) uses the same definitions and was cross-checked against this implementation on identical row sets.

### Testing

`pytest tests/unit/metrics/test_metrics_passk.py` — 33 tests: estimator known values and edge cases, invalid-count rejection, cross-run grouping, doc-id dedupe, null handling, custom k/threshold, CLI smoke tests. `ruff check` / `ruff format` clean.

```release-note
Added pass@k / pass^k metric calculators and a rows.json summarizer CLI (`python -m devops_bench.metrics.passk`).
```

Part of #92

### CODE REVIEW

File: devops_bench/metrics/passk.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    )
+    args = parser.parse_args(argv)
+
+    out_path = (Path(args.out) if args.out else Path(args.root) / _OUT_FILENAME).resolve()
+
+    files = discover_row_files(args.root, exclude=(out_path,))
+    if not files:
+        parser.error(f"no rows.json files found under {args.root}")
+
+    summary = summarize_rows(load_rows(files), k=args.k, threshold=args.threshold)
```
[the remark this answers is not in the document]
Comment (isadominguez314) [reply devops_bench/metrics/passk.py]: Fixed. Excluding only ROOT/rows.json misses the combined files inside each run root when the CLI scans a parent directory, so it now skips any rows.json with a sibling manifests.json if its per-task inputs are also present.

**Comment by isadominguez314** [comment]
@coderabbitai review

File: devops_bench/metrics/passk.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+
+def _drop_combined(files: list[Path]) -> list[Path]:
+    """Drop aggregate outputs (``manifests.json`` sibling) whose per-task inputs are also present."""
+    return [
+        f
+        for f in files
+        if not (
+            (f.parent / "manifests.json").exists()
+            and any(f.parent in o.parents for o in files if o != f)
```
[the remark this answers is not in the document]
Comment (isadominguez314) [reply devops_bench/metrics/passk.py]: Fixed. Aggregate outputs are now skipped whenever any per-task rows are found, wherever they live. A scan of only aggregate outputs still keeps them

**Comment by isadominguez314** [comment]
@coderabbitai review

File: devops_bench/metrics/passk.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    return {"k": k, "threshold": threshold, "setups": setups}
+
+
+_OUT_FILENAME = "passk_summary.json"
+
+
+def _drop_combined(files: list[Path]) -> list[Path]:
+    """Drop aggregate outputs (``manifests.json`` sibling) if any per-task rows are present."""
+    per_task = [f for f in files if not (f.parent / "manifests.json").exists()]
+    return per_task or files
```
[the remark this answers is not in the document]
Comment (isadominguez314) [reply devops_bench/metrics/passk.py]: Won't fix, known trade-off. Rows don't record which run an aggregate came from, and matching on setup/task/iteration would also drop legitimate repeated runs, which are exactly what pass@k pools. Scanning a mix of per-task runs and aggregate-only runs isn't supported. An explicit --exclude flag can be added once this CLI has callers.

**Comment by isadominguez314** [comment]
@coderabbitai review
